documentation/modules/exploit/multi/http/spip_porte_plume_previsu_rce.md

```diff
@@ -0,0 +1,152 @@
+## Vulnerable Application
+
+This Metasploit module exploits a Remote Code Execution vulnerability in SPIP versions up to and including 4.2.12.
+The vulnerability occurs in SPIP’s templating system where it incorrectly handles user-supplied input, allowing an attacker
+to inject and execute arbitrary PHP code.
+This can be achieved by crafting a payload that manipulates the templating data processed by the `echappe_retour()` function,
+which invokes `traitements_previsu_php_modeles_eval()`, containing an `eval()` call.
+
+To replicate a vulnerable environment for testing:
+
+1. Install SPIP using the provided Docker Compose configuration.
+2. Use the image `ipeos/spip:4.2.12` to ensure the environment is vulnerable.
+3. Verify that the SPIP instance is accessible on the local network.
+
+### Docker Setup
+
+Use the following Docker Compose file to set up the environment:
+
+```yaml
+version: '3.8'
+
+services:
+  db:
+    image: mariadb:10.5
+    restart: always
+    environment:
+      - MYSQL_ROOT_PASSWORD=MysqlRootPassword
+      - MYSQL_DATABASE=spip
+      - MYSQL_USER=spip
+      - MYSQL_PASSWORD=spip
+    networks:
+      - spip-network
+
+  app:
+    image: ipeos/spip:4.2.12
+    restart: always
+    depends_on:
+      - db
+    environment:
+      - SPIP_AUTO_INSTALL=1
+      - SPIP_DB_SERVER=db
+      - SPIP_DB_LOGIN=spip
+      - SPIP_DB_PASS=spip
+      - SPIP_DB_NAME=spip
+      - SPIP_SITE_ADDRESS=http://localhost:8880
+    ports:
+      - 8880:80
+    networks:
+      - spip-network
+
+networks:
+  spip-network:
+    driver: bridge
+```
```

Comment:
does this docker compose config work you for you? all I end up with is this page
<img width="655" alt="image" src="https://github.com/user-attachments/assets/671a7720-d4df-4226-ae22-52fff87ea924">

same for the non docker instructions later on too actually

Reply:
ah, after you do either docker or the other install method you need to navigate to `/ecrire` to perform the install, could that be added to the docs please?

Reply:
that still didn't work with the docker compose setup but I got the exploit working with the standard install after following the install steps at `/ecrire` and then rebooting the app

Reply:
Hello, yes I will add that in the documentation. And fix the docker compose. Here mariadb is even useless since we can do the config with SQLite.